modules/mobile/app/controllers/mobile/v0/contact_info_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Mobile
+  module V0
+    class ContactInfoController < ApplicationController
+      before_action :map_logingov_to_idme
+      after_action :handle_vet360_id, only: :show
+
+      def show
+        render json: Mobile::V0::ContactInfoSerializer.new(@current_user, @current_user.vet360_contact_info)
+      end
+
+      private
+
+      # solution so old app versions will still treat LOGINGOV accounts as multifactor
+      def map_logingov_to_idme
+        if @current_user.identity.sign_in[:service_name].include? 'LOGINGOV'
+          @current_user.identity.sign_in[:service_name] = 'oauth_IDME'
+        end
+      end
+
+      def handle_vet360_id
```

Comment:
This is another thing I hadn't really thought about. The reason this was in the users show method is that user show is hit right after login, so we treat it as a proxy for login. This new endpoint won't be hit on login because contact info is only used on specific screens so we'll wait to make the call until we need it. 

I can understand why you'd want to move it here, since this is dependent upon vet360 id, but I'm not sure we should move it for a couple of reasons. One is that if we do it on signin, that may actually allow the link to occur before the user requests contact info, which would allow this endpoint to provide more data. Another is that we may switch over to the new endpoints one by one with the user endpoint switch being last. That could lead to a situation where this happens twice in the user serializer then here. So my instinct is to keep this in the user endpoint instead.

Reply:
I'm good with that. As you said, this dependency on vet360 id made me think it was worth considering having hear but I'm good with removing it